### PR TITLE
[SFARAP-25] Embed assemblies in generated jar

### PR DESCRIPTION
**Pull request by duncanpMS**

### CODE REVIEW

File: RoslynPluginGenerator/AnalyzerPluginGenerator.cs [hunk 1]
```diff
@@ -94,15 +91,22 @@ public bool Generate(NuGetReference analyzeRef, string language, string sqaleFil
...
 
                 // Collect the remaining data required to build the plugin
+                RoslynPluginDefinition definition = new RoslynPluginDefinition();
+                definition.Language = language;
+                definition.SqaleFilePath = sqaleFilePath;
                 definition.PackageId = package.Id;
                 definition.PackageVersion = package.Version.ToString();
                 definition.Manifest = CreatePluginDefinition(package);
                 definition.RulesFilePath = Path.Combine(baseDirectory, "rules.xml");
-                definition.EmbeddedZipFileName = package.Id + ".zip"; // TODO - generate the zip file
```
Comment (duncanpMS) [inline RoslynPluginGenerator/AnalyzerPluginGenerator.cs]: The zip file is being created as follows: the local NuGet repository has unzipped the files to a single directory. I've changed the NuGet handler interface to return this directory, and we just zip it again.
However, the zip contains the original .nupkg file, which is in itself a zip file. I'll change this is a later commit.

File: RoslynPluginGenerator/NuGet/NuGetPackageHandler.cs [hunk 2]
```diff
@@ -38,6 +39,10 @@ public NuGetPackageHandler(IPackageRepository remoteRepository, string localCach
             this.logger = logger;
             this.remoteRepository = remoteRepository;
             this.localCacheRoot = localCacheRoot;
+
+            Directory.CreateDirectory(localCacheRoot);
+            this.packageManager = new PackageManager(remoteRepository, localCacheRoot);
+            this.packageManager.Logger = new NuGetLoggerAdapter(logger);
```
Comment (duncanpMS) [inline RoslynPluginGenerator/NuGet/NuGetPackageHandler.cs]: Before this interface only had one method so we didn't need to keep hold of the manager instance. Now it's needed for both methods.

File: Tests/Common/ZipFileChecker.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
+//-----------------------------------------------------------------------
+// <copyright file="ZipFileChecker.cs" company="SonarSource SA and Microsoft Corporation">
```
Comment (duncanpMS) [inline Tests/Common/ZipFileChecker.cs]: Existing helper class: moved and renamed from _JarChecker_ to _ZipFileChecker_.
I've changed all of refs in the class from "jar" to "zip".

**Comment by bgavrilMS** [comment]
LGTM

**Comment by HSAR** [comment]
LGTMT

**Pull request closed without merging**